java/client/src/main/java/glide/api/models/configuration/BaseSubscriptionConfiguration.java

```diff
@@ -87,6 +90,10 @@ protected <M extends ChannelMode> void addSubscription(
          * @param context The {@link #context}.
          */
         public B callback(MessageCallback callback, Object context) {
+            if (context != null && callback == null) {
```

Comment:
I think you can just ust `@Nonnull` instead for both args

Reply:
Otherwise a god of code coverage will punish us

Reply:
I didn't do this intentionally because the user might want to disable their callback.

Reply:
I put this in the documentation to make this more explicit.